Automatically update now and next.

mstratford opened the pull request:
Improvements still to make:

- [x] Update status of message box.
- [x] Update just after the hour and half hour, not just every 30mins.

mstratford committed:
Need full jquery for Ajax.

mstratford committed:
Basic updating now & next.

mstratford commented:
Closes #41

Danzibob committed:
Add (en/dis)abling of message box
(untested)

Danzibob committed:
Add basic "Jump to current show" functionality
Soon to be tied into the auto-updating

Danzibob committed:
Make C&N and jumping to a show in schedule more robust

Danzibob committed:
Have jump button appear only on this week

Danzibob committed:
Merge branch 'brooke-schedule-key' into mstratford-updating-nowNext

Danzibob committed:
Added emphasis animation after scroll link
Turns out this isn't at all related to C&N anymore. whoops.

Danzibob committed:
Import C&N JS with the template

Danzibob committed:
Make C&N update just after the half hour

Danzibob committed:
Merge branch 'development' into mstratford-updating-nowNext

Danzibob committed:
Honestly codacy, the one time I want you to chastise me

Danzibob committed:
The great appeasement
Also fix jump to current show when nothing's on air

Danzibob committed:
The second great appeasement

Danzibob commented:
There's still a couple of complexity issues (which seem entirely valid but I don't know what to do about them) and one "generic item sink" which I think is a non-issue. Otherwise, ready for review

Was also considering adding a bar in the style `(M|T|W|T|F|S|S)` above / next to the jump button to link to each day, to further enhance mobile-view. Thoughts?

I thought this button would be far more tied into the C&N updater this PR was for, but it turned out not to be at all. Sorry Matt :p

Danzibob committed:
Silly useless semicolons

LordAro commented on public/js/schedule.js:
```diff
@@ -0,0 +1,45 @@
+    }
+  }
+  hour = zpad(hour,2);
+  weekday = daysOfWeek[weekday];
```
variable redeclaration/reuse - should be avoided

LordAro reviewed (commented):
codacy's complexity threshold can probably be raised - we use 10 at work

Otherwise fine - just check your coding style - lots of missing spaces before `{` and after `,`

LordAro commented on views/partials/base.tmpl:
```diff
@@ -53,6 +53,7 @@
 {{block "footer-block" .}}
     {{template "footer" .}}
 {{end}}
+<script>var MyRadioAPIKey = {{.PageContext.MyRadioAPIKey }};</script>
```
Not sure this should be a thing. i believe at present the api key is not exposed anywhere? (the config in this repo excepted) Maybe there should be a "public apikey" config variable? That said of course, everything on the site is "public", so maybe it doesn't matter

Brookke replied:
I believe this same notation is used elsewhere

mstratford commented:
From what I remember, this API key is (or at least should be) the different public one only capable of doing end user side stuff (ie loading current and next, maybe sending in messages depending on how that's implemented).

We do need to do a clear out of the API keys / permissions, and have a "production website key" instead. The one in the config should probably be revoked.

Regards,
Matthew Stratford,
Web developer, AV technician & all round techy guy.
https://mstratford.net
On 21 Jun 2018, 20:29 +0100, Brooke Hatton , wrote:
> @Brookke commented on this pull request.
> In views/partials/base.tmpl:
> > @@ -53,6 +53,7 @@
> {{block "footer-block" .}}
>     {{template "footer" .}}
> {{end}}
> +<script>var MyRadioAPIKey = {{.PageContext.MyRadioAPIKey }};</script>
> I believe this same notation is used elsewhere
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Danzibob committed:
Run JS beautifier and change text to match the Go output

Danzibob committed:
Beautify and fix variable re-use

Danzibob committed:
Merge branch 'development' into mstratford-updating-nowNext

Danzibob commented:
With regards to the API key, I'm inclined to agree with @LordAro. ~~As is, the API key exposed in the JS is the same as that in .myradio.key~~, and it's only being used for a single endpoint. This notation _was_ used elsewhere, but has since been removed after messages were moved back-end.

~~I'm starting a new branch to implement this change, and ideally we can merge that before this.~~
EDIT: Matt's already inserting a seperate key from config.toml, all I have to do is mess about with making a new key for public JS

Danzibob committed:
Differentiate public API key
Added new API key to database with required permissions
(required permissions being literally none)

Danzibob committed:
Change text again so it fits in the box; minor appeasement

Danzibob committed:
Merge branch 'mstratford-updating-nowNext' of https://github.com/UniversityRadioYork/2016-site into mstratford-updating-nowNext

Danzibob commented:
Added new public API key to the database and to the config file. Ready to merge.
Config on server _should_ need updating but probably doesn't until we clear out the old API key.

mstratford committed:
Merge branch 'development' into mstratford-updating-nowNext

Brookke commented on views/schedule_week.tmpl:
```diff
@@ -122,7 +130,13 @@
+<!-- Jump to day button -->
+<script>var StartHour = {{.PageData.StartHour}};</script>
+<script src='{{url "/js/schedule.js"}}' id="jumpToNow"></script>
+{{end}}
```
Needs trailing line

Brookke commented on views/schedule_week.tmpl:
```diff
@@ -122,7 +130,13 @@
     </div>
   </div>
 </div>
+
```
Probably could remove this empty line

Brookke reviewed (approved):
If it works then looks good to me other than a few erroneous lines/missing lines

Danzibob committed:
Fix empty lines / lack of; center jump button

mstratford committed:
Fix now next update scheduling

mstratford reviewed (commented):
Seems good, seems to work changing my computer clock around.

mstratford committed:
Fix Semicolon

mstratford committed:
Fix hours to 2 digits too (to match HTML)

mstratford committed:
Remove extra line.

LordAro commented on config.toml.example:
```diff
@@ -30,7 +30,8 @@
 	newsTwitter = "URYNews"
 
 	odName = "On Tap"
-	myRadioAPIKey = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```
has this key been revoked?

mstratford replied:
Has now...

LordAro commented on public/js/index.js:
```diff
@@ -22,3 +22,4 @@ function onGoogleLoad() {
     });
   });
 }
+
```
unnecessary newline

LordAro commented on views/elements/current_and_next.tmpl:
```diff
@@ -66,6 +68,8 @@
 {{end}}
+</div>
 {{end}}
+
```
also unnecessary newline

mstratford committed:
Remove empty line.

mstratford committed:
Remove another new line.

mstratford commented:
Note #184 changes some of the text layout for the now and next.

Brookke commented on public/js/currentAndNext.js:
```diff
@@ -0,0 +1,121 @@
+            // Default case (regular show)
+            return "<h5 class=\"ellipsis\">" +
+              show.title +
+              "</h5>" +
```
All of the headings in this will now need updating to match the new headings

mstratford replied:
👍

mstratford committed:
Merge branch 'development' into mstratford-updating-nowNext

mstratford committed:
Update for accessibility PR changes.

mstratford committed:
Merge remote-tracking branch 'origin/brooke-podcast-api-patch' into mstratford-updating-nowNext

mstratford committed:
Fixes for #187

mstratford committed:
Few more html fixes.

mstratford committed:
(Hopefully) fix Codacy

mstratford committed:
Fix Variable Assigned to Object Injection Sink

mstratford committed:
Merge remote-tracking branch 'origin/development' into mstratford-updating-nowNext

mstratford commented:
Right, OK. I've fixed all the Codacy issues (bar the Complexity, which was recommended to change to 10 in the settings). Plz merge ❤️
